docs/management/api/upload-api.asciidoc

```diff
@@ -0,0 +1,100 @@
+[[upload-api]]
+=== Upload file to host
+
+Upload a file to the host running {elastic-defend}.
+
+You must have the `File Operations` Security Kibana privilege as part of your role and at least an Enterprise license to perform this action.
+
+==== Request URL
+
+`POST <kibana host>:<port>/api/endpoint/action/upload`
+
+The request must include:
+
+* the `Content-Type: multipart/form-data` HTTP header.
+
+
+==== Request body
+
+A `multipart/form-data` with the following:
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`endpoint_ids` |Array (String) |The IDs of endpoints where you want to issue this action. |Yes
+|`alert_ids` |Array (String) |If this action is associated with any alerts, they can be specified here. The action will be logged in any cases associated with the specified alerts. |No
+|`case_ids` |Array (String) |The IDs of cases where the action taken will be logged. |No
+|`comment` |String |Attach a comment to this action's log. The comment text will appear in associated cases. |No
+|`parameters.overwrite` |Boolean |Overwrite file on host if it already exists. |No
+|`file` |Stream |The file content to be uploaded. |Yes
+|==============================================
+
+
+===== Example requests
+
+Upload a file named `fix-malware.sh` to a host with an `endpoint_id` value of `ed518850-681a-4d60-bb98-e22640cae2a8`. It assumes that the file is located in the same directory from where the command is being entered:
+
+
+[source,shell]
+--------------------------------------------------
+curl -X POST "api/endpoint/action/upload" \
+-H 'kbn-xsrf: true' \
+-H 'Content-Type: multipart/form-data' \
+-F 'endpoint_ids: ["ed518850-681a-4d60-bb98-e22640cae2a8"]' \
+-F "file=@fix-malware.sh" <1>
+--------------------------------------------------
+<1> The relative link to the file to be uploaded
```

Comment:
Is the `@` symbol part of the required syntax? We might want to make that more explicit, because otherwise it seems unexpected to me, since most key-value pairs use a single character as a connector, like `file=filename.sh` or `file:filename.sh`.

```suggestion
-F "file=@fix-malware.sh" <1>
--------------------------------------------------
<1> The relative path to the file to upload. Note the path must be preceded with `@`.
```